### DISCUSSION TITLE
Feature: Content Collections

### DISCUSSION THREAD

**Discussion by austinm911** [TanStack/router#3047] [Ideas]
Haven't seen this mentioned so creating a discussion to track this feature.

Overall it seems that Velite and Content Collections are currently well-maintained and worth exploring.

- https://github.com/timlrx/contentlayer2 - fork of Content Layer, which is now unmaintained. Many of the libraries below originated as alternatives to Content Layer
	- Some alternatives are listed here: https://github.com/contentlayerdev/contentlayer/discussions/628
- https://github.com/sdorra/content-collections
	- Used by [Magic UI](https://github.com/magicuidesign/magicui/blob/f69220cc275f6f0fa86b2196edc1918bcd94ea36/content-collections.ts#L4)
	- Has adapters for Vite, SolidStart, and others.	
	- see https://sdorra.dev/posts/2024-01-15-content-collections
- https://github.com/zce/velite
	- used Chakra UI 
	- https://www.mikevpeeren.nl/blog/refactoring-contentlayer-to-velite
	- https://scriptbenders.com/folio/building-static-site-with-blog-nextjs-react-radix-shadcn-ui/00900-setting-up-velite/
	- https://nooc.me/en/posts/integrate-a-blog-in-nextjs-with-velite
- https://mdxjs.com/docs/getting-started/#vite

**Comment by ally-ahmed** [comment]
A blog/portfolio starter template using content collections.
Source
https://github.com/ally-ahmed/tss-blog-starter
Demo
https://tss-blog-starter.pages.dev/
